client/js/gdpr/chart.js

```diff
@@ -79,289 +90,323 @@ export function createChart(container, histories, dates) {
-      );
+    if (
+      this.hoverState.seriesIndex !== null &&
+      this.hoverState.seriesIndex !== hitIndex
+    ) {
+      this.clearHover();
     }
 
-    if (hit === null) {
-      hoverState.series = null;
-      hoverState.x = null;
-      hoverState.y = null;
-    } else {
-      const seriesEntry = historyEntries[hit];
-      const seriesElement = seriesElements[hit];
-      seriesElement.dispatchEvent(
-          new CustomEvent('series-hit', {
-            detail: {
-              x,
-              y,
-              key: seriesEntry[0],
-            },
-            bubbles: true,
-          }),
-      );
-
-      hoverState.series = hit;
-      hoverState.x = x;
-      hoverState.y = y;
+    if (hitIndex >= 0) {
+      this.setHover(x, y, hitIndex);
     }
-  }, {passive: true, capture: true});
+  }
 
-  const clearHover = () => {
-    if (hoverState.series !== null) {
-      const seriesElement = seriesElements[hoverState.series];
-      seriesElement.dispatchEvent(
-          new CustomEvent('series-clear', {bubbles: true}),
+  onMouseLeave() {
+    this.clearHover();
+  }
+
+  onScroll() {
+    this.clearHover();
+  }
+
+  setHover(x, y, seriesIndex) {
+    const key = this.rows[0][seriesIndex];
+    const seriesEl = this.svg.querySelectorAll('.series')[seriesIndex];
+
+    seriesEl.dispatchEvent(
+        new CustomEvent('series-hit', {
+          detail: {
+            x,
+            y,
+            key,
+          },
+          bubbles: true,
+          composed: true,
+        }),
+    );
+
+    this.hoverState.seriesIndex = seriesIndex;
+    this.hoverState.x = x;
+    this.hoverState.y = y;
+  }
+
+  clearHover() {
+    const {seriesIndex} = this.hoverState;
+
+    if (seriesIndex !== null) {
+      const seriesEl = this.svg.querySelectorAll('.series')[seriesIndex];
+      seriesEl.dispatchEvent(
+          new CustomEvent('series-clear', {bubbles: true, composed: true}),
       );
     }
 
-    hoverState.series = null;
-    hoverState.x = null;
-    hoverState.y = null;
-  };
+    this.hoverState.seriesIndex = null;
+    this.hoverState.x = null;
+    this.hoverState.y = null;
+  }
 
-  svg.addEventListener('mouseleave', () => clearHover());
-  scrollContainer.addEventListener('scroll', () => clearHover());
+  createChart() {
+    const svg = document.createElementNS(SVG_NS, 'svg');
+    svg.setAttribute('class', 'chart');
+    svg.setAttribute(
+        'viewBox',
+        // shift down so that lines are vertically centered
+        `0 ${RUN_SCALE_Y * 0.5} ` +
+        `${this.dates.length * RUN_SCALE_X} ${(NUM_POSITIONS + 0.5) * RUN_SCALE_Y}`,
+    );
+    svg.append(this.createDefs());
+    svg.append(this.createGrid());
+
+    const seriesContainer = document.createElementNS(SVG_NS, 'g');
+
+    /** @type {SVGGElement[]} */
+    const seriesElements = [];
+
+    let index = 0;
+
+    for (const history of this.histories.values()) {
+      const hue = index * 15 % 360;
```

Comment:
I think this was addressed in an earlier PR, but add a comment explaining that you want hues to be 15 degrees apart on the color wheel so we have context as to what 15 and 360 here stand for.

Reply:
argh why does it keep disappearing

Reply:
fixed